lib/nerves_hub_link/configurators/local_cert_key.ex

```diff
@@ -0,0 +1,62 @@
+defmodule NervesHubLink.Configurator.LocalCertKey do
+  @behaviour NervesHubLink.Configurator
+
+  alias NervesHubLink.{Certificate, Configurator.Config}
+
+  @cert_kv_path "nerves_hub_cert"
+  @key_kv_path "nerves_hub_key"
+
+  @impl NervesHubLink.Configurator
+  def build(%Config{} = config) do
+    config
+    |> maybe_add_cacerts()
+    |> maybe_add_cert()
+    |> maybe_add_key()
+    |> maybe_add_sni()
+  end
+
+  defp maybe_add_cacerts(%{ssl: ssl} = config) do
+    %{config | ssl: Keyword.put_new(ssl, :cacerts, Certificate.ca_certs())}
+  end
+
+  defp maybe_add_cert(%{ssl: ssl} = config) do
+    ssl =
+      cond do
+        ssl[:cert] || ssl[:certfile] ->
+          ssl
+
+        cert = Nerves.Runtime.KV.get(@cert_kv_path) ->
+          cert = Certificate.pem_to_der(cert)
+
+          Keyword.put(ssl, :cert, cert)
+
+        true ->
+          Keyword.put_new(ssl, :certfile, "#{config.data_path}/cert.pem")
```

Comment:
```suggestion
          Keyword.put_new(ssl, :certfile, Path.join(config.data_path, "cert.pem")
```
Probably worth using the path helper here just for cleanliness so we don't end up with double slashes (`/data/nh//cert.pem`, etc)